MoviesExample/app/src/main/java/com/novoda/demo/movies/MoviesAdapter.java

```diff
@@ -81,44 +79,37 @@ public long getItemId(final int position) {
 
     class MovieItem extends RecyclerView.ViewHolder {
 
-        @BindView(R.id.movie_item_title) TextView text;
-        @BindView(R.id.movie_item_poster) ImageView poster;
-        @BindView(R.id.movie_item_rating) TextView rating;
-
-        MovieItem(View itemView, final Listener listener) {
-            super(itemView);
-            ButterKnife.bind(this, itemView);
-            itemView.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(final View v) {
-                    int adapterPosition = getAdapterPosition();
-                    if (adapterPosition != -1) {
-                        listener.onMovieSelected(moviesSate.get(adapterPosition));
-                    }
-                }
-            });
+        MoviesListItemBinding binding;
+
+        MovieItem(MoviesListItemBinding binding, final Listener listener) {
+            super(binding.getRoot());
+            this.binding = binding;
+            binding.setListener(listener);
+
         }
 
         public void bind(final Movie movie) {
-            text.setText(movie.title);
-            rating.setText(Double.toString(movie.rating));
-            Glide.with(itemView.getContext()).load(movie.posterUrl()).into(poster);
+            binding.setMovie(movie);
+            binding.executePendingBindings();
         }
     }
 
     static class LoadPageItem extends RecyclerView.ViewHolder {
 
-        @BindView(R.id.movie_item_title) TextView text;
+        MoviesListCardBinding binding;
         Listener listener;
 
-        LoadPageItem(View itemView, Listener listener) {
-            super(itemView);
-            ButterKnife.bind(this, itemView);
+        LoadPageItem(MoviesListCardBinding binding, Listener listener) {
+            super(binding.getRoot());
+            this.binding = binding;
             this.listener = listener;
         }
 
         public void bind(final int page) {
-            text.setText("Loading page " + page);
+            Movie movie = new Movie();
+            movie.title = "Loading page " + page;
+
+            binding.setMovie(movie);
```

Comment:
This is a hack: the loading footer item is a `Movie` item with the title "Loading page X" and no image.
It was like that in the original app and I didn't want to change the behavior.

Reply:
Could this also be done using a separate `LoadingBinding` or something similar?